### ISSUE TITLE
Detect when bot was removed from the room?

### ISSUE THREAD

**Issue by hems** [telegraf/telegraf#846]
## Feature Request

I would like to know when my bot was removed from a group chat

### Proposal:

We should have an event just like `new_chat_members ` and `left_chat_member` ? 

### Current behavior:

Currently i can check for "new_chat_members" and see if the `id` is the same id from the bot, but using the botInfo property.

But when the bot is removed i'm not receiving a `left_chat_member` event?

### Desired behavior:

`left_chat_member` to be executed when the bot is removed from the room

### Use case: 

We're creating a bot that can be added and removed to groups, would be good to monitor how many groups the bot is currently active.

Of course, other features could be created on top of this little feature, but currently our only requirement is to monitor which rooms the bot is participating.

**Comment by TiagoDanin** [comment]
This will happen using Telegraf or not.
Event `left_chat_member` is only triggered in `groups`, `new_chat_members` are triggered in `supergroups` and `groups`

> https://core.telegram.org/bots/api#message
> new_chat_members | Array of User | Optional. New members that were added to the group or supergroup and information about th
> -- | -- | --
> 
> left_chat_member | User | Optional. A member was removed from the group, information about them (this member may be the bot itself)
> -- | -- | --

**Comment by hems** [comment]
> This will happen using Telegraf or not.
> Event `left_chat_member` is only triggered in `groups`, `new_chat_members` are triggered in `supergroups` and `groups`
> 
> > https://core.telegram.org/bots/api#message
> > 
> > 
> > 
> > new_chat_members
> > Array of User
> > Optional. New members that were added to the group or supergroup and information about th
> > 
> > 
> > 
> > 
> > 
> > 
> > left_chat_member
> > User
> > Optional. A member was removed from the group, information about them (this member may be the bot itself)

thank for your clarification.

is there any known workarounds for left_chat_member in supergroups?

**Comment by hems** [comment]
TLDR: is there a reliable way to detect when a bot was removed from a room?

**Comment by KnorpelSenf** [comment]
I'm pretty sure this is not possible. However, if you bot is in a sufficiently large group, you can probably infer that the bot was removed based on the timestamp of the last message in that particular chat.

**Comment by ksaitor** [comment]
so far this has worked for me:

```js
const bot = new Telegraf('apikey…')
bot.use(async (ctx, next) => {
  const my_chat_member = ctx.update.my_chat_member
  if (my_chat_member) {
    if (['kicked', 'left'].includes(my_chat_member.new_chat_member.status)) {
      // bot was kicked
    } else if (
      ['member', 'administrator'].includes(my_chat_member.new_chat_member.status)
    ) {
      // bot was added to a group or channel
    }
  }
  next()
})
```

**Comment by KnorpelSenf** [comment]
Yes, these types of updates were introduced by Telegram in the meantime.
